File: ExercismFSharp/ExercismFSharp/PerfectNumbers.fs
module PerfectNumbers

open System

type NumberType =
    | Perfect
    | Abundant
    | Deficient

let classify num =
    let factorSum =
        [1 .. num - 1]
        |> Seq.filter (fun n -> num % n = 0)
        |> Seq.sum

    let diff = factorSum - num

    match diff with
    | _ when diff < 0 -> NumberType.Deficient
    | 0 -> NumberType.Perfect
    | _ when diff > 0 -> NumberType.Abundant
    | _ -> raise (Exception "unreachable code")

File: Exercism/python/perfect-numbers/perfect_numbers.py
import math


def classify(number):
    if number <= 0:
        raise ValueError("Number must be positive")

    factor_sum = sum(filter(lambda x: number % x == 0, range(1, math.floor(number / 2) + 1)))

    if factor_sum > number:
        return "abundant"
    elif factor_sum < number:
        return "deficient"
    else:
        return "perfect"
